### ISSUE TITLE
ubuntu16.04无法运行lantern

### ISSUE THREAD

**Issue by yfszzx** [getlantern/lantern#4605]
ubuntu16.04 32位，安装lantern-installer-beta-32-bit.deb没有报错，但运行lantern显示如下错误:
/usr/lib/lantern/lantern-binary: OK
DEBUG flashlight.ui: ui.go:50 Creating tarfs filesystem that prefers local resources at /lantern/src/github.com/getlantern/lantern-ui/app
DEBUG flashlight.ui: ui.go:50 Creating tarfs filesystem that prefers local resources at /lantern/src/github.com/getlantern/lantern-ui/app
Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused

(lantern:7977): Gtk-WARNING **: cannot open display:
求指点

**Comment by wfjsw** [comment]
Your PC don't have a GUI screen? 
Compile Lantern from source by `HEADLESS make linux-386`. Also https://github.com/getlantern/lantern/wiki/Installation:-server

**Comment by yfszzx** [comment]
I have shutted  down  GUI screen. I will try .Thank you

**Comment by yfszzx** [comment]
I compiled Lantern from source by HEADLESS=true make linux-386,but I don't know what's wrong whick my PC :
** VERSION was not set, using default version. This is OK while in development.
Build tags: headless
Extra ldflags:
# github.com/getlantern/flashlight/client
src/github.com/getlantern/flashlight/client/reverseproxy.go:21: unknown http.Transport field 'MaxIdleTime' in struct literal
# github.com/getlantern/flashlight/proxied
src/github.com/getlantern/flashlight/proxied/proxied.go:391: tr.MaxIdleTime undefined (type *http.Transport has no field or method MaxIdleTime)
src/github.com/getlantern/flashlight/proxied/proxied.go:392: tr.EnforceMaxIdleTime undefined (type *http.Transport has no field or method EnforceMaxIdleTime)
Makefile:238: recipe for target 'linux-386' failed
make: *** [linux-386] Error 2

**Comment by iamlook** [comment]
@yfszzx met the same problem as you too. Did you find the solution?

**Comment by iamlook** [comment]
@yfszzx have to use lantern's go-lang fork: https://github.com/getlantern/go , net/http is different from master branch.

**Comment by wfjsw** [comment]
That's dirty...
Time to fire up Docker.

**Comment by wfjsw** [comment]
@yfszzx @iamlook Use commit `6ad15e1614c6a2140f5d8ad7e573868e782dbe9d` instead of `devel`. It is last-known-good.

**Comment by yfszzx** [comment]
@iamlook I have try lantern's golang fork, but compile failed,I don't know reason. Can you use lantern in terminal mod now?
@iamlook I have try  commit 6ad15e1614c6a2140f5d8ad7e573868e782dbe9d.I don't know how to clone the commit,so I download zip file.But it is also wrong:
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
Makefile:194: recipe for target 'src/github.com/getlantern/flashlight/ui/resources.go' failed
make: *** [src/github.com/getlantern/flashlight/ui/resources.go] Error 128

**Comment by wfjsw** [comment]
@yfszzx
```
git clone https://github.com/getlantern/lantern.git
cd lantern
git checkout 6ad15e1614c6a2140f5d8ad7e573868e782dbe9d
```

**Comment by yfszzx** [comment]
@wfjsw I change my OS as ubuntu server 16.04.It also does not  work.error：
Missing "node" command.
Makefile:312: recipe for target 'require-node' failed
make: *** [require-node] Error 1
commit 6ad15e1614c6a2140f5d8ad7e573868e782dbe9d also does not work

It's make me crazy

**Comment by wfjsw** [comment]
@yfszzx
Run 
```
curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
sudo apt-get install -y nodejs
```
there is nothing serious.

P.S.
`6ad15e1614c6a2140f5d8ad7e573868e782dbe9d` has a nasty bug of gc. You may restart it every a few hours.
Remember currently if you want to build lantern from HEAD you have to install docker to make things easier.
If you are still experiencing problem I can send you thre binary if necessary.

**Comment by wfjsw** [comment]
P.P.S:
next time quote your log with ```

**Comment by krast** [comment]
PLS try docker run lantern

apt-get docker
service docker start or systemctl start docker

docker pull krast0/lantern
docker run --name lantern -d -p 8787:8787 krast0/lantern
curl -x 'http://127.0.0.1:8787' -v https://www.google.com

On Jun 12, 2016, at 1:48 AM, 不会画画的中医不是好程序员 <notifications@github.com<mailto:notifications@github.com>> wrote:


@wfjsw<https://github.com/wfjsw> I change my OS as ubuntu server 16.04.It also does not work.error：
Missing "node" command.
Makefile:312: recipe for target 'require-node' failed
make: *** [require-node] Error 1
commit 6ad15e1<https://github.com/getlantern/lantern/commit/6ad15e1614c6a2140f5d8ad7e573868e782dbe9d> also does not work

It's make me crazy

**Comment by yfszzx** [comment]
@wfjsw  can you send me a binary file? thank you so much. My OS is ubuntu server 16.04 64bit.My email:yfszzx@gmail.com

**Comment by yfszzx** [comment]
@krast thank you, I have do what I can do, but I still can't compile , I give up.

**Comment by iamlook** [comment]
@yfszzx Yes, I can use lantern in terminal mod now. I built Lantern with HEADLESS=true . every thing works fine now :)
you need install an official go-lang first, then build the lantern's fork version, after that, change the PATH to the bin directory that you built.
And you need install the other prerequisites package as [https://github.com/getlantern/lantern#prerequisites](https://github.com/getlantern/lantern#prerequisites) , such as Nodejs & NPM. 

Then your compilation should be ok.

**Comment by JdkingSilence** [comment]
@iamlook  Is there any documentation about how to "build the lantern's fork version" ? I repalced $GOROOT (official version) with go-fork, but it dosent work...

**Comment by Boshin** [comment]
@wfjsw 
我的平台macOS Sierra， go version go1.9 darwin/amd64，
编译lantern的6ad15e1614c6a2140f5d8ad7e573868e782dbe9d目前是OK的，但是编译devel branch报了如下错误
uilding development lantern
** VERSION was not set, using default version. This is OK while in development.
Build tags:
Extra ldflags:
# github.com/getlantern/flashlight/proxied
src/github.com/getlantern/flashlight/proxied/proxied.go:391:5: tr.MaxIdleTime undefined (type *http.Transport has no field or method MaxIdleTime)
src/github.com/getlantern/flashlight/proxied/proxied.go:392:5: tr.EnforceMaxIdleTime undefined (type *http.Transport has no field or method EnforceMaxIdleTime)
# github.com/getlantern/flashlight/client
src/github.com/getlantern/flashlight/client/reverseproxy.go:20:14: unknown field 'MaxIdleTime' in struct literal of type http.Transport
src/github.com/getlantern/flashlight/client/reverseproxy.go:22:11: transport.EnforceMaxIdleTime undefined (type *http.Transport has no field or method EnforceMaxIdleTime)
make: *** [lantern] Error 2

请教一下这个问题的原因是什么？ go的版本不对？ 我clone了 lantern给的go版本但是编译不过，无论是go1.4还是用go1.9， 目前不知道该怎么弄了。

**Comment by wfjsw** [comment]
@Boshin 用readme.md给的那个custom fork of go
（现在编译lantern怕是毫无用处）

**Comment by Boshin** [comment]
我把涉及MaxIdleTime的代码给注释掉了，能编译通过也能用，不过不知道有什么bug没有。那么现在该怎么编译呢？

**Comment by Boshin** [comment]
@wfjsw

**Comment by wfjsw** [comment]
@Boshin 搞maxidletime是为了防止开一堆没用的链接最后导致too many open files
我也不知道后边怎么办了 你自己摸索着试试呗（

**Comment by AiPick** [comment]
@Boshin 我是这样编译的，不会报错。不要将VERSION放在环境变量中，而是直接作为编译选项。

$ sudo VERSION=9.9.9 make package-linux-amd64
>........
>........
>Created package {:path=>"lantern-manoto_9.9.9_amd64.deb"}
-> lantern_9.9.9_amd64.deb

**Comment by Boshin** [comment]
@AiPick 具体列举一下，你的平台是mac os不， 我最近把系统重装了发现lantern不能用了?  我晚上把log贴出来。

**Comment by zach007** [comment]
@wfjsw 大神，有时间出一个教程，手把手在ubuntu上安装lantern，无图形界面，我看stack overflow上有人问，而且在issus有很多相似的，但是官方文档实在是太简洁了就是说：HEADLESS = true，我断断续续弄了一个月没搞明白，查了很久谷歌，愣是没弄明白，请为记得为小白答疑！万分感谢了！

**Comment by wfjsw** [comment]
@zach007 问题是现在开源版的lantern已经近一年没更新了，新版全都是闭源的，所以继续折腾这个并没有多大意义。
